### PR TITLE
11 28 updates

### PR DESCRIPTION
**Pull request by TianL520**

### CODE REVIEW

File: Management/Clients/AirQualityDataClient.cs
Comment (TianL520) [inline Management/Clients/AirQualityDataClient.cs:108]: Now we directly use the static map instead of calling supported cities from the API.

**Comment by MakinoAlan** [comment]
Don't merge it until I finish reviewing it and sign off

File: Management/DomainModels/AbbrToStateDict.cs
Comment (MakinoAlan) [inline Management/DomainModels/AbbrToStateDict.cs:1]: Please consider consolidate this one and another dictionary class to the following static class:

```
using Management.DomainModels;
using System.Collections.Generic;

namespace Management
{
    public static class AbbreviationToState
    {
        private static readonly Dictionary<string, State> _stateLookUp = new Dictionary<string, State>
        {
            { "CA", State.Wrap("California") },
            { "GA", State.Wrap("Georgia") },
            { "IL", State.Wrap("Illinois") },
            { "MA", State.Wrap("Massachusetts") },
            { "WA", State.Wrap("Washington") },
        };

        private static readonly Dictionary<string, List<City>> _cityLookUp = new Dictionary<string, List<City>>()
        {
            { "CA", new List<City>() { City.Wrap("Los Angeles"), City.Wrap("San Diego"), City.Wrap("San Jose"), City.Wrap("San Francisco"), City.Wrap("Fresno") } },
            { "GA", new List<City>() { City.Wrap("Atlanta"), City.Wrap("Savannah"), City.Wrap("Athens"), City.Wrap("Roswell"), City.Wrap("Marietta") } },
            { "IL", new List<City>() { City.Wrap("Chicago"), City.Wrap("Naperville"), City.Wrap("Joliet"), City.Wrap("Rockford"), City.Wrap("Springfield") } },
            { "MA", new List<City>() { City.Wrap("Boston"), City.Wrap("Worcester"), City.Wrap("Springfield"), City.Wrap("Brockton"), City.Wrap("Quincy") } },
            { "WA", new List<City>() { City.Wrap("Seattle"), City.Wrap("Spokane"), City.Wrap("Tacoma"), City.Wrap("Vancouver"), City.Wrap("Bellevue") } },
        };

        public static State GetStateFullName(string abbreviation)
            => _stateLookUp[abbreviation];

        public static IEnumerable<City> GetSupportedCities(string countryCode)
            => _cityLookUp[countryCode];
    }
}
```

Reason: 

1. You are not accessing any instance, the data you are trying to access is "fixed"
2. Inilize instace in multiple class will create a lot lot of overhead, and is not necessary

File: Management/Clients/AirQualityDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/AirQualityDataClient.cs:101]: Why are you changing the logic here?

**It is calling the api to get list of supported cities**, now after your change it's calling the fixed data.

File: Management/Clients/AirQualityDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/AirQualityDataClient.cs:169]: why you are changing the edge case here? 

So now 100, 150, 200 and 300 will be a overlap case, which is what I tried to avoid

File: Management/Clients/CovidDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/CovidDataClient.cs:46]: nit: string interpolation

```
v1/states/{state.Value.ToLower()}/current.json"

File: Management/Clients/CovidDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/CovidDataClient.cs:54]: oh this API has the limit call issue as well? LOL

File: Management/Clients/CovidDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/CovidDataClient.cs:58]: is 99999999 too much? in terms of fake the positive case in a State.

File: Management/Clients/CovidDataClient.cs
Comment (MakinoAlan) [inline Management/Clients/CovidDataClient.cs:77]: 1. You are not using the Parllel execution, so you do not need to use concurrentBag.
2. You are repeating yourself here. GetSingleStateAsync will return only 1 TUPLE, and you add into the list then calculate the average of for list of 1.....

Suggestsion: change at line 85 to public async Task<(State, double)> CalculateScoreForStateAsync. That's it

Comment (MakinoAlan) [reply Management/Clients/CovidDataClient.cs:77]: I did not see the change

File: Management/Clients/CovidDataClient.cs
Comment (MakinoAlan) [reply Management/Clients/CovidDataClient.cs:58]: Same here, did you forget to push your commit

File: Management/Clients/AirQualityDataClient.cs
Comment (TianL520) [reply Management/Clients/AirQualityDataClient.cs:101]: Hope you can see my comments now. I have restored the logic back :)

File: Management/Clients/CovidDataClient.cs
Comment (TianL520) [reply Management/Clients/CovidDataClient.cs]: Fixed.

File: Management/Clients/CovidDataClient.cs
Comment (TianL520) [reply Management/Clients/CovidDataClient.cs]: Removed because found out that the current covid API doesn't have limits.

File: Management/Clients/AirQualityDataClient.cs
Comment (TianL520) [reply Management/Clients/AirQualityDataClient.cs]: Restored.

File: Test/Management/Integration/AirQualityDataClientIntegrationTests.cs
Comment (TianL520) [inline Test/Management/Integration/AirQualityDataClientIntegrationTests.cs:52]: Apparently, the air quality API that we are using also has a daily limit (I'm not sure how many exactly). Thus, I kept two API keys in here just in case we need more testings without failing caused by the daily limit.

File: Management/DecisionEngine.cs
Comment (MakinoAlan) [inline Management/DecisionEngine.cs:51]: I think the agreement is 10?

File: Management/DecisionEngine.cs
Comment (MakinoAlan) [inline Management/DecisionEngine.cs:151]: Well this works, but I still think the weighted score system make more sense. As in the real world different factor will be weighted differently

File: Management/Interface/ICovidDataClient.cs
Comment (MakinoAlan) [inline Management/Interface/ICovidDataClient.cs:19]: You dont need it, as it is defined in parent interface

File: Service/appsettings.Development.json
Comment (MakinoAlan) [inline Service/appsettings.Development.json:11]: why comment out not just remove it?

File: Test/Management/Integration/AirQualityDataClientIntegrationTests.cs
Comment (MakinoAlan) [reply Test/Management/Integration/AirQualityDataClientIntegrationTests.cs:52]: Ideally intergration should only run a few times - during CI, usually.
I will take a deeper look, and mark this skip for local. Integration tests with third party api should not be called a lot.

**Review by MakinoAlan — APPROVED** [review]
Please take a look at my comment and feel free to merge.
In general it's good, thanks you for the hard work!

File: Management/DecisionEngine.cs
Comment (Alying) [reply Management/DecisionEngine.cs:51]: yes, ideally! @TianL520

File: Service/appsettings.Development.json
Comment (TianL520) [reply Service/appsettings.Development.json:11]: Just in case we run out of daily limit calls for internal testing purposes. In the final product, we will only keep one and remove the comment.

File: Test/Management/Integration/AirQualityDataClientIntegrationTests.cs
Comment (TianL520) [reply Test/Management/Integration/AirQualityDataClientIntegrationTests.cs:52]: Thanks!

File: Management/DecisionEngine.cs
Comment (TianL520) [reply Management/DecisionEngine.cs:151]: We can definitely keep the weighted score system in. :)

**Pull request merged**
